auxiliary/new_auto_mapper.py

```diff
@@ -0,0 +1,40 @@
+import cv2
+
+from auxiliary.cv2_handler import CV2Handler
+from auxiliary.homographic_handler import HomographicHandler
+from auxiliary.image_handler import ImageHandler
+from auxiliary.model_interpreter import ModelInterpreter
+
+
+class AutoMapper:
```

Comment:
Thinking further direction which we might have more than homographic transform feature, I would suggest we either rename this class name to be specific on Homographic functionality or more generic name such that this class will play a role as a toggle for multiple features (homographic, heatmap, people_analyzer). I would think of an abstract/base class that has predefined methods and the overlapping utilities between features. And for each feature, it need to inherited from this base class and implement these 3 overlapping methods or additional methods specific to the features.

As reference maybe we also think of homographic transformation as a single task instead of feature. So heat map extraction will be another task. Thus we can create a base class `TaskTemplate` and all the other tasks must inherited from this class. For reference, I always get inspired by huggingface, check this https://github.com/huggingface/datasets/blob/2.2.1/src/datasets/tasks/base.py

example : 

```python
from abc import ABC, abstractmethod

class TaskTemplate(ABC):
    def __init__(self, **kwargs):
        self.model_interpreter = kwargs.get("model_interpreter")
        self.image_handler = kwargs.get("image_handler")
        self.homographic_handler = kwargs.get("homographic_handler")
        self.cv2_handler = kwargs.get("cv2_handler")

    @abstractmethod
    def generate_raw_outputs(self):
        pass

    @abstractmethod
    def stream_using_cv2(self):
        pass

    @abstractmethod
    def stream_as_image(self):
        pass
```

Reply:
This means that we're creating an abstract class, which is quite similar to an Interface to achieve extensibility within the app. 

A nice idea, I think it will be a lot easier to move forward when we know that the app is extensible up to a certain degree. However, I not previously created something like that before, so I might need your help a bit to know that we're aiming for the same goal.